content/operate/rs/references/rest-api/requests/bdbs/actions/recover.md

```diff
@@ -0,0 +1,147 @@
+---
+Title: Recover database requests
+alwaysopen: false
+categories:
+- docs
+- operate
+- rs
+description: REST API requests for database recovery
+headerRange: '[1-2]'
+linkTitle: recover
+weight: $weight
+---
+
+| Method | Path | Description |
+|--------|------|-------------|
+| [GET](#get-bdbs-actions-recover) | `/v1/bdbs/{uid}/actions/recover` | Get database recovery plan  |
+| [POST](#post-bdbs-actions-recover) | `/v1/bdbs/{uid}/actions/recover` | Recover database  |
+
+## Get recovery plan {#get-bdbs-actions-recover}
+
+```sh
+GET /v1/bdbs/{int: uid}/actions/recover
+```
+
+Fetches the recovery plan for a database. The recovery plan provides information about the recovery status, such as whether recovery is possible, and details on available files to use for recovery.
+
+#### Required permissions
+
+| Permission name | Roles |
+|-----------------|-------|
+| [view_bdb_recovery_plan]({{< relref "/operate/rs/references/rest-api/permissions#view_bdb_recovery_plan" >}}) | admin<br />cluster_member<br />cluster_viewer<br />db_member<br />db_viewer |
+
+### Request {#get-request}
+
+#### Example HTTP request
+
+```sh
+GET /bdbs/1/actions/recover
+```
+
+#### URL parameters
+
+| Field | Type | Description |
+|-------|------|-------------|
+| uid | integer | The unique ID of the database. |
+
+### Response {#get-response}
+
+Returns a JSON object that represents the database's recovery plan, including recovery files and status.
+
+#### Example response body
+
+```json
+{
+  "data_files": [
+    {
+      "filename": "appendonly-1.aof",
+      "node_uid": "1",
+      "shard_slots": "1-2048"
+    },
+    {
+      "filename": "appendonly-2.aof",
+      "node_uid": "2",
+      "shard_slots": "2049-4096"
+    }
+  ],
+  "status": "ready"
+}
+```
+
+#### Status codes {#get-status-codes}
+
+| Code | Description |
+|------|-------------|
+| [200 OK](http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.2.1) | No error. |
+| [404 Not Found](http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.5) | Database UID does not exist. |
```

Comment:
The original pages show a message saying they have been superseded. Suggested URLs are the new replacements (the new URLs are actually used in links lower in the page).
```suggestion
| [200 OK](https://www.rfc-editor.org/rfc/rfc9110.html#name-200-ok) | No error. |
| [404 Not Found](https://www.rfc-editor.org/rfc/rfc9110.html#name-404-not-found) | Database UID does not exist. |
```